frontend/src/scenes/data-warehouse/DataWarehouseScene.tsx

```diff
@@ -1,43 +1,358 @@
+import { useMemo, useState, useEffect } from 'react'
 import { SceneExport } from 'scenes/sceneTypes'
+import { PipelineTab } from '~/types'
 import { FEATURE_FLAGS } from 'lib/constants'
 import { featureFlagLogic } from 'lib/logic/featureFlagLogic'
 import { useValues } from 'kea'
 import { NotFound } from 'lib/components/NotFound'
+import { urls } from 'scenes/urls'
+import { LemonButton, LemonCard, LemonTag, Tooltip } from '@posthog/lemon-ui'
+import { PaginationControl, usePagination } from 'lib/lemon-ui/PaginationControl'
+import { IconPlusSmall, IconCheckCircle, IconInfo } from '@posthog/icons'
+import { IconOpenInNew } from 'lib/lemon-ui/icons'
+import { dataWarehouseSettingsLogic } from './settings/dataWarehouseSettingsLogic'
+import { dataWarehouseSceneLogic } from './settings/dataWarehouseSceneLogic'
+import { TZLabel } from 'lib/components/TZLabel'
+import { IconCancel, IconSync, IconExclamation, IconRadioButtonUnchecked } from 'lib/lemon-ui/icons'
+import { fetchRecentActivity, fetchTotalRowsProcessed, type UnifiedRecentActivity } from './externalDataSourcesLogic'
 
-export const scene: SceneExport = {
-    component: DataWarehouseScene,
+export const scene: SceneExport = { component: DataWarehouseScene }
+
+const LIST_SIZE = 5
+
+const getSourceType = (sourceType: string, availableSources?: Record<string, any> | null): 'Database' | 'API' => {
+    const fields = availableSources?.[sourceType]?.fields || []
+    if (fields.some((f: any) => f.name === 'connection_string' || ['host', 'port', 'database'].includes(f.name))) {
+        return 'Database'
+    }
+    if (fields.some((f: any) => f.type === 'oauth' || ['api_key', 'access_token'].includes(f.name))) {
+        return 'API'
+    }
+    return 'API'
+}
+
+interface DashboardDataSource {
+    id: string
+    name: string
+    type: 'Database' | 'API'
+    status: string | null
+    lastSync: string | null
+    rowCount: number | null
+    url: string
 }
 
 export function DataWarehouseScene(): JSX.Element {
```

Comment:
I would suggest trying to use lemontables rather than this card format. The cards can work but because these are pretty straightforward lists I would expect them to be pretty close to the existing table formats. I could be wrong here but I'd try the tables first